'issue17'

eliseev-eag opened the pull request:
Created AnalyticsAgent - class, which provides API for starting task for photo recognition by AnalyticsClient.

green124b-dev committed:
Added interface IAnalitycsClient and class AnalyticsClient

green124b-dev committed:
It was added models for recognitioin results rescponce and added communication with server

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
It was fixed bug connected with incompatibility of dependencies(Microsoft.AspNet.WebApi.Client)

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
Created model for recognition request response

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
Added class for testing AnalyticsClient

green124b-dev committed:
Fixed typo in return type of Analytics client method(UploadPhoto)

green124b-dev committed:
Fixed handling of bad request response in AnalyticsClient methods

green124b-dev committed:
Added converting photo to Base64 for transfering

green124b-dev committed:
Added integration testing for AnalyticsAgent

eliseev-eag committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into 'issue17'

eliseev-eag committed:
added AnalyticsAgent
wrote first version of function which take photo from queue and send to analyticsClient

eliseev-eag committed:
Merge branch 'issue21' of https://github.com/alex-ks/backend-comptech into 'issue17'

eliseev-eag committed:
Added PollingTask, CancelPolling and Dispose methods on AnalyticsAgent
Added public methods to AnalyticsAgent which provides API to start task to recognition photo and cancel this task

green124b-dev committed:
Fixed typos

DVEfremov commented on Comptech.Backend/src/Comptech.Backend.Service/Analytics/AnalyticsAgent.cs:
```diff
@@ -0,0 +1,106 @@
+                Thread.Sleep(pollingTimeout);
+            }
+            
+            Photo photo = photoRepository.GetPhotoById(recognitionTask.PhotoId);
```
get методы обычно должны гарантированно возвращать результат или эксепшен здес  так же ?

DVEfremov reviewed (commented):
в остальном лукс гуд ту ми

sumboid commented on Comptech.Backend/src/Comptech.Backend.Service/Analytics/AnalyticsAgent.cs:
```diff
@@ -0,0 +1,106 @@
+            
+            Photo photo = photoRepository.GetPhotoById(recognitionTask.PhotoId);
+
+            bool isPhotoAploaded = false;
```
Опечатка: isPhotoUploaded

sumboid commented on Comptech.Backend/src/Comptech.Backend.Service/Analytics/AnalyticsAgent.cs:
```diff
@@ -0,0 +1,106 @@
+            Photo photo = photoRepository.GetPhotoById(recognitionTask.PhotoId);
+
+            bool isPhotoAploaded = false;
+            while (!isPhotoAploaded)
```
На самом деле, сомнительный момент -- что будет, если analytics сервис будет недоступен? Мы ведь рискуем попасть в бесконечный цикл на этом моменте. А если прибавить достаточно большое число пользователей -- не приведет ли это к нехорошим последствиям для бекенда?

eliseev-eag replied:
Я понимаю, что это не очень хорошо, но у меня нет опыта в написании подобных вещей. Как лучше поступить, задать timeout ожидания, ограничить кол-во попыток или использовать какой-то другой подход?

sumboid replied:
Да, я думаю, что ограничить число попыток какой-нибудь константой будет нормальным решением.

eliseev-eag replied:
спасибо

sumboid commented on Comptech.Backend/src/Comptech.Backend.Service/Analytics/AnalyticsAgent.cs:
```diff
@@ -0,0 +1,106 @@
+            }
+
+            RecognitionResults recognitionResults = null;
+            while (recognitionResults == null)
```
Аналогичный вопрос про бесконечный цикл здесь

green124b-dev committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
It was improved code style

green124b-dev committed:
Fixed typo

green124b-dev committed:
Merge branch 'issue21' of https://github.com/alex-ks/backend-comptech into issue21

eliseev-eag committed:
fixed infinite loop when photo uploading
added param PhotoUploadTryCount - number of attempts to upload photo
rename isPhotoAploaded variable to isPhotoUploaded

eliseev-eag committed:
added write recognitionResults to IRecognitionResultsRepository

eliseev-eag committed:
added test to ideal scenario

eliseev-eag committed:
refactor test

green124b-dev committed:
Fixed typos im urls of requests

green124b-dev committed:
Merge branch 'issue21' of https://github.com/alex-ks/backend-comptech into issue21

green124b-dev committed:
It was fixed bug in deserialization

alex-ks committed:
Merge pull request #34 from alex-ks/issue21

Implemented AnalyticsClient

eliseev-eag committed:
add singleton instance AnalyticsAgent to configuration

eliseev-eag committed:
Merge branch 'master' of https://github.com/alex-ks/backend-comptech into 'issue17'

eliseev-eag committed:
AnalyticsAgent added to DI; Added agent request to UploadPhotoAndStartSession method (to create singleton)
